tests/ops/op_math/test_sum.py

```diff
@@ -1275,6 +1286,133 @@ def test_adjoint(self):
+
+        # check with qml.dot
+        op1 = qml.dot(coeffs, obs, grouping_type="qwc")
+        assert op1.grouping_indices == ((0, 1), (2,))
+
+        # check with qml.sum
+        sprods = [qml.s_prod(c, o) for c, o in zip(coeffs, obs)]
+        op2 = qml.sum(*sprods, grouping_type="qwc")
+        assert op2.grouping_indices == ((0, 1), (2,))
+
+        # check with Sum directly
+        op3 = Sum(*sprods, grouping_type="qwc")
+        assert op3.grouping_indices == ((0, 1), (2,))
+
+    def test_grouping_is_correct_compute_grouping(self):
+        """Basic test checking that grouping with compute_grouping works as expected"""
+        a = qml.PauliX(0)
+        b = qml.PauliX(1)
+        c = qml.PauliZ(0)
+        obs = [a, b, c]
+        coeffs = [1.0, 2.0, 3.0]
+
+        op = qml.dot(coeffs, obs, grouping_type=None)
+        assert op.grouping_indices is None
+
+        op.compute_grouping(grouping_type="qwc")
+        assert op.grouping_indices == ((0, 1), (2,))
+
+    def test_grouping_does_not_alter_queue(self):
+        """Tests that grouping is invisible to the queue."""
+        a = qml.PauliX(0)
+        b = qml.PauliX(1)
+        c = qml.PauliZ(0)
+        obs = [a, b, c]
+        coeffs = [1.0, 2.0, 3.0]
+
+        with qml.queuing.AnnotatedQueue() as q:
+            op = qml.dot(coeffs, obs, grouping_type="qwc")
+
+        assert q.queue == [op]
+
+    def test_grouping_for_non_groupable_sums(self):
+        """Test that grouping is computed correctly, even if no observables commute"""
+        a = qml.PauliX(0)
+        b = qml.PauliY(0)
+        c = qml.PauliZ(0)
+        obs = [a, b, c]
+        coeffs = [1.0, 2.0, 3.0]
+
+        op = qml.dot(coeffs, obs, grouping_type="qwc")
+        assert op.grouping_indices == ((0,), (1,), (2,))
+
+    def test_grouping_method_can_be_set(self):
+        """Tests that the grouping method can be controlled by kwargs.
+        This is done by changing from default to 'rlf' and checking the result."""
+        a = qml.PauliX(0)
+        b = qml.PauliX(1)
+        c = qml.PauliZ(0)
+        obs = [a, b, c]
+        coeffs = [1.0, 2.0, 3.0]
+
+        # compute grouping during construction with qml.dot
+        op1 = qml.dot(coeffs, obs, grouping_type="qwc", method="lf")
+        assert op1.grouping_indices == ((2, 1), (0,))
+
+        # compute grouping during construction with qml.sum
+        sprods = [qml.s_prod(c, o) for c, o in zip(coeffs, obs)]
+        op2 = qml.sum(*sprods, grouping_type="qwc", method="lf")
+        assert op2.grouping_indices == ((2, 1), (0,))
+
+        # compute grouping during construction with Sum
+        op3 = Sum(*sprods, grouping_type="qwc", method="lf")
+        assert op3.grouping_indices == ((2, 1), (0,))
+
+        # compute grouping separately
+        op4 = qml.dot(coeffs, obs, grouping_type=None)
+        op4.compute_grouping(method="lf")
+        assert op4.grouping_indices == ((2, 1), (0,))
+
+    def test_grouping_type_can_be_set(self):
```

Comment:
Perhaps also one with `"commuting"` (e.g. `X(0) @ X(1)` and `Y(0) Y(1)` commute but dont qubit-wise-commute)